conda_build/environ.py

```diff
@@ -976,18 +995,53 @@ def get_install_actions(
                 if not any(
                     re.match(r"^%s(?:$|[\s=].*)" % pkg, str(dep)) for dep in specs
                 ):
-                    actions["LINK"] = [
-                        spec for spec in actions["LINK"] if spec.name != pkg
+                    actions[LINK_ACTION] = [
+                        prec for prec in actions[LINK_ACTION] if prec.name != pkg
                     ]
         utils.trim_empty_keys(actions)
         cached_actions[(specs, env, subdir, channel_urls, disable_pip)] = actions.copy()
         last_index_ts = index_ts
     return actions
 
 
+def get_package_records(
+    prefix,
+    specs,
+    env,
+    retries=0,
+    subdir=None,
+    verbose=True,
+    debug=False,
+    locking=True,
+    bldpkgs_dirs=None,
+    timeout=900,
+    disable_pip=False,
+    max_env_retry=3,
+    output_folder=None,
+    channel_urls=None,
+):
+    return get_install_actions(
```

Comment:
Having the direction from "new function"->"old function" here lets us avoid too many internal changes for now.
(In the next version, this function will of course just hold the actual code.)

Reply:
This will trigger the deprecation warning during runtime though, even though it shouldn't when being called as `get_package_records`.

Please extract the bulk of the `get_install_actions` function into a separate “private” function, which can be called from both `get_install_actions` (decorated for deprecation) and `get_package_records` (without deprecation).

Reply:
Ah, good catch!
I've put the function definition now into `get_package_records` with just unwrapping/wrapping the `actions` dictionary to keep changes minimal for now.